numpyro/distributions/distribution.py

```diff
@@ -350,6 +350,45 @@ def variance(self):
         return np.broadcast_to(self.base_dist.variance, self.batch_shape + self.event_shape)
 
 
+class ImproperUniform(Distribution):
+    """
+    A helper distribution with zero :meth:`log_prob` and improper
```

Comment:
nit: "improper sample" -> "arbitrarily distributed sample" or something. The term 'improper' specifically means "non normalized and possibly non-nonrmalizable" as in "improper uniform prior over the real line".

Reply:
Thanks for explaining! I'll make the change.

Re `sample` method: I will raise not ImplementationError here as you did. It is better.